### ISSUE TITLE
Double View (allows you to view both new messages & old messages at the same time)

### ISSUE THREAD

**Issue by Terraman2** [Vencord/plugin-requests#399]
### Plugin Description

Such a plugin would be very useful because sometimes I want to get caught up on old messages while also staying active in the conversation. The way it should work is there should be a double view button & a double view keyboard shortcut, & if you activate double view, then the bottom half of the normal message view would be replaced with a view of the newest messages.

### Request Agreement

- [X] I read the above and made sure my plugin is possible and allowed
- [X] I checked the [existing plugins](https://vencord.dev/plugins) and made sure my plugin doesn't already exist
- [X] I [searched the existing issues](https://github.com/Vencord/plugin-requests/issues?q=is%3Aissue) and made sure a similar request doesn't already exist

**Comment by mynameisbob1928** [comment]
So you basically want to be able to view 2 channels at the same time. (potentially the same channel twice) Ngl I would use that

**Comment by Terraman2** [comment]
> So you basically want to be able to view 2 channels at the same time. (potentially the same channel twice) Ngl I would use that

No, not 2 channels at the same time, still 1 channel at a time but new messages & old messages at the same time.

**Comment by Terraman2** [comment]
> So you basically want to be able to view 2 channels at the same time. (potentially the same channel twice) Ngl I would use that

I didn't expect someone to reply so fast. Since you did, now I need to make sure that everyone knows about the last sentence that I edited into the request so that the button will be in the right place.

**Comment by mynameisbob1928** [comment]
2 different channels though would be nice. And this would still support your original idea

**Comment by Terraman2** [comment]
> [mynameisbob1928](/mynameisbob1928)

Well, that would make it more complicated for something that not everyone wants, so that should be optional.

**Comment by Terraman2** [comment]
> 2 different channels though would be nice. And this would still support your original idea

Well, that would make it more complicated for something that not everyone wants, so that should be optional.

**Comment by Terraman2** [comment]
> 2 different channels though would be nice. And this would still support your original idea

That wouldn't fully support my original idea, because with my original idea, when switch channels, the new message view would automatically switch to showing the new messages of that channel, but with that version, you would have to manually change it.

**Comment by Terraman2** [comment]
& I realized that instead of the double view being separated vertically, it should work the same as the forum post view. The text on the top should be replaced with "New Messages" when Sticky View is disabled, & the name of the channel when Sticky View is enabled. Clicking the more button should show an Open in Full View button, a Sticky View button, a Switch to Current Channel button that's only there when Sticky View is enabled, a Search button, & a Pins button.

**Comment by Terraman2** [comment]
> & I realized that instead of the double view being separated vertically, it should work the same as the forum post view. The text on the top should be replaced with "New Messages" when Sticky View is disabled, & the name of the channel when Sticky View is enabled. Clicking the more button should show an Open in Full View button, a Sticky View button, a Switch to Current Channel button that's only there when Sticky View is enabled, a Search button, & a Pins button.

I realized that actually the horizontally split view is better for using 2 channels at once but worse for viewing old messages while staying active in the conversation, so there should actually be 2 types of split view.

**Comment by Terraman2** [comment]
> > & I realized that instead of the double view being separated vertically, it should work the same as the forum post view. The text on the top should be replaced with "New Messages" when Sticky View is disabled, & the name of the channel when Sticky View is enabled. Clicking the more button should show an Open in Full View button, a Sticky View button, a Switch to Current Channel button that's only there when Sticky View is enabled, a Search button, & a Pins button.
> 
> I realized that actually the horizontally split view is better for using 2 channels at once but worse for viewing old messages while staying active in the conversation, so there should actually be 2 types of split view.

Actually, because of this, the split view for being in 2 places at once doesn't really work, so I'll just change everything to pretty much the original version & copy & paste what's currently there into a comment.

**Comment by Terraman2** [comment]
Here is the description of the edited plugin idea that I realized wouldn't really work:

Such a plugin would be very useful because sometimes I want to get caught up on old messages while also staying active in the conversation. 

There would be a Show New Messages button just to the left of the Hide Member List button, along with an option Show New Messages keyboard shortcut, & enabling Show New messages would make it so that bottom half of the message view always shows the newest messages no matter where you're scrolled to.

The pre-existing forums split view would be repurposed so that it can also show the messages you're currently viewing in the current channel/thread/forum-post so you can scroll away from them & still have them on-screen, & there would be a split view button on the top right to the left of the threads button. When you switch channels/threads/forum-posts with split view enabled, the side view would switch to showing the messages there & it would start at the same scroll position that you're at in the main view just like the first time.

There would be a "Sticky View" button that makes the split view show the same messages when you go to a different channel/thread/forum, & when sticky view is enabled, there would also be a button called "Switch to Current Messages" or something, that manually changes the view to the current channel/thread/forum. There would be a configuration option to have sticky view enabled by default, but that wouldn't affect the pre-existing forums split view because if the pre-existing forums split view had sticky view by default, that would result in inconvenience rather than convenience due to the pre-existing forums split view being specifically for viewing forum messages & forum posts at the same time, & not for viewing multiple sets of messages at once.

When sticky view is enabled, opening a forum post should immediately open it in the main view to prevent it from overriding the current side view, & to prevent that change from causing inconvenience, forum posts should always have a close button when you're in split view mode, even if they're in the main view, & if the post is in the main view, the close button would change the main view to the forum channel.

**Comment by Terraman2** [comment]
Actually, the problem was just that reusing the forums split view was a bad idea. The split view should be vertical.

**Comment by Terraman2** [comment]
> So you basically want to be able to view 2 channels at the same time. (potentially the same channel twice) Ngl I would use that

I made lots of bad edits to the idea but I fixed it & I think the plugin should probably just be exactly what it was originally going to be. The only potential difference would be the option to be able to take the sticky new messages to another channel & have the message box send messages in the channel of the sticky new messages rather than the channel you're taking them to. But while I like that change, I have no idea how it could be executed, so it might have to not be included just for that reason, at least until someone figures it out.

**Comment by Terraman2** [comment]
> So you basically want to be able to view 2 channels at the same time. (potentially the same channel twice) Ngl I would use that

I made some more edits & I think now it's flawless. But someone still needs to figure out some way to get the reading new messages & sending messages in 1 channel but reading other messages in another channel at the same time thing to work.

**Comment by Dodge100** [comment]
This seems to be similar to the [BD Channel Tabs](https://betterdiscord.app/plugin/ChannelTabs) plugin.

**Comment by mynameisbob1928** [comment]
> This seems to be similar to the [BD Channel Tabs](https://betterdiscord.app/plugin/ChannelTabs) plugin.

Not quite, channel tabs from bd is keeping easy access to two or more different channels/servers. What is being requested here is something to view the same channel twice (i added on that you could view two different channels at the same time)

**Comment by Dodge100** [comment]
> > This seems to be similar to the [BD Channel Tabs](https://betterdiscord.app/plugin/ChannelTabs) plugin.
> 
> Not quite, channel tabs from bd is keeping easy access to two or more different channels/servers. What is being requested here is something to view the same channel twice (i added on that you could view two different channels at the same time)

ah, I see. I would think that this would be easily accomplished with 2 renderings of discord side by side.
